modules/product/src/main/java/com/opengamma/strata/product/swap/type/FixedInflationSwapTemplate.java

```diff
@@ -0,0 +1,463 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product.swap.type;
+
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.time.Period;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.ImmutableValidator;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.strata.basics.BuySell;
+import com.opengamma.strata.basics.market.ReferenceData;
+import com.opengamma.strata.basics.market.ReferenceDataNotFoundException;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.product.TradeTemplate;
+import com.opengamma.strata.product.swap.SwapTrade;
+
+/**
+ * A template for creating Inflation swap trades.
+ * <p>
+ * This defines almost all the data necessary to create a Inflation single currency {@link SwapTrade}.
+ * The trade date, end date, lag, notional and fixed rate are required to complete the template and create the trade.
+ * As such, it is often possible to get a market price for a trade based on the template.
+ * The market price is typically quoted as a bid/ask on the fixed rate.
+ */
+@BeanDefinition
+public final class FixedInflationSwapTemplate
+    implements TradeTemplate, ImmutableBean, Serializable {
+
+  /**
+   * The period between the spot value date and the start date.
+   * <p>
+   * This is often zero, but can be greater if the swap if <i>forward starting</i>.
+   * This must not be negative.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final Period periodToStart;
```

Comment:
Template should have a "tenor" (like "FixedIborSwapTemplate"). No periodToStart is required for inflation swap (forward starting inflation swaps require a different mechanism, better to avoid them at this stage).